src/lib.rs

```diff
@@ -1,6 +1,560 @@
+use core::panic;
+
 #[doc = include_str!("../README.md")]
 
-/// Say hi to the entire world :)
-pub fn greet() {
-	println!("Hello World!");
+#[derive(PartialEq, Eq)]
+#[derive(Debug)]
+pub enum StringDiffOpKind {
+    Substitute(char, char),  
+    Insert(char),
+    Delete(char),
+    Transpose(usize, usize),
+}
+
+#[derive(Debug)]
+pub struct StringDiffOp {
+    pub kind: StringDiffOpKind,
+    pub index: isize,
+}
```

Comment:
There should be a constructor for this type, which should look like:
```rs
impl StringDiffOp {
   pub fn new(kind: StringDiffOpKind, index: isize) {
       Self { kind, index }
   }
}
```

And any instance of creating a new `StringDiffOp` should use this, via `StringDiffOp::new()`